### DISCUSSION TITLE
Restricting Kademlia queries

### DISCUSSION THREAD

**Discussion by nazar-pc** [libp2p/rust-libp2p#2475] [Q&A]
I'm fairly new to libp2p and wondering what would be correct way to restrict some queries (both values and content providers) since I'll be using alternative approach for those.

Would something like dropping request in Kademlia wrapper in `inject_event()` be sufficient or maybe this is not a supported use case at all?

**Comment by mxinden** [comment]
Not supported today. In case you just don't want to pay the cost of storage, the below hack would work for you:

https://github.com/mxinden/rust-libp2p-server/blob/7b630eed7187341817b5d781babfbdee02e41f9d/src/behaviour.rs#L32-L36

Otherwise implementing your own record store?

**Reply by nazar-pc** [reply]
I already have a custom record store that doesn't actually store anything, I was just wondering if there is a better way to stop it earlier somehow.

The server is nice as an example to look at, thanks!
